src/position.cpp

```diff
@@ -52,10 +52,10 @@ namespace Zobrist {
 
 namespace {
 
-const string PieceToChar(" PNBRQKHE pnbrqkhe");
+const string PieceToChar(" PNBRHEQK pnbrheqk");
 
-const Piece Pieces[] = { W_PAWN, W_KNIGHT, W_BISHOP, W_ROOK, W_QUEEN, W_KING, W_HAWK, W_ELEPHANT,
-                         B_PAWN, B_KNIGHT, B_BISHOP, B_ROOK, B_QUEEN, B_KING, B_HAWK, B_ELEPHANT };
+const Piece Pieces[] = { W_PAWN, W_KNIGHT, W_BISHOP, W_ROOK, W_HAWK, W_ELEPHANT, W_QUEEN, W_KING,
+                         B_PAWN, B_KNIGHT, B_BISHOP, B_ROOK, B_HAWK, B_ELEPHANT, B_QUEEN, B_KING };
 
 // min_attacker() is a helper function used by see_ge() to locate the least
```

Comment:
Line 70 could also be simplified now, I think. [It does not seem to be possible to add comments on a line in segments where there is no code change, that is why I am commenting here.]

Reply:
Ha ha. I definitely made that change at some point but it obviously didn't make it into the PR. Kind of funny that the complicated version works just the same as "Pt + 1" now.

All of your other points are valid too. I will make the changes tomorrow.